features/weth/Weth.tsx

```diff
@@ -114,33 +132,199 @@ const Weth = () => {
+                  </IconAndNameContainer>
+                  <TokenBalance>Ξ {prettyEthBalance}</TokenBalance>
+                </BalanceElement>
+                <BalanceElement>
+                  <IconAndNameContainer>
+                    <TokenIcon src={ETHEREUM_LOGO_URL} />
+                    <TokenName>WETH</TokenName>
+                  </IconAndNameContainer>
+                  <TokenBalance>Ξ {prettyWethBalance}</TokenBalance>
+                </BalanceElement>
+              </Container>
+            </Box>
+          </Grid>
+          <Grid item md={6} sm={6} xs={12}>
+            <Box>
+              <Container>
+                <Typography variant="h5">Wrap & Unwrap</Typography>
+                <InputElement>
+                  <form
+                    style={{ width: "100%" }}
+                    noValidate
+                    autoComplete="off"
+                    onSubmit={(e) => handleClick(e, ACTION_TYPE.WRAP)}
+                  >
+                    <TextField
+                      fullWidth
+                      type="number"
+                      label="ETH"
+                      value={ethAmount}
+                      onChange={(e) => setEthAmount(e.target.value)}
+                      variant="outlined"
+                      inputProps={{ min: "0", max: ethBalance as string }}
+                      error={ethAmountAboveBalance}
+                      helperText={
+                        ethAmountAboveBalance && `Your ETH balance is too low`
+                      }
+                      InputLabelProps={{
+                        shrink: true,
+                      }}
+                      InputProps={{
+                        endAdornment: (
+                          <InputAdornment position="end">
+                            <Button
+                              type="submit"
+                              variant="contained"
+                              disabled={
+                                ethAmountAboveBalance || ethAmountToWrap <= 0
+                              }
+                            >
+                              Wrap
+                            </Button>
+                          </InputAdornment>
+                        ),
+                      }}
+                    />
+                  </form>
+                </InputElement>
+                <InputElement>
+                  <form
+                    style={{ width: "100%" }}
+                    noValidate
+                    autoComplete="off"
+                    onSubmit={(e) => handleClick(e, ACTION_TYPE.UNWRAP)}
+                  >
+                    <TextField
+                      fullWidth
+                      type="number"
+                      label="WETH"
+                      value={wethAmount}
+                      onChange={(e) => setWethAmount(e.target.value)}
+                      variant="outlined"
+                      inputProps={{ min: "0", max: wethBalance as string }}
```

Comment:
+1 on setting the max to the ethBalance. that's pretty useful. One question though: do we want to restrict this input field? I agree 100% with disabling the button but from a ux perspective do you want to _block_ the user from interacting with this field? might be cleaner to let them enter any wrap/unwrap button but if it's bigger than their balance then just let it error out with the `error` modifier and `helperText`. Therefore my suggestion is to remove this `inputProps max` field.

Reply:
Why would a user ever want to wrap/unwrap more than their balance? it would never succeed right?

Reply:
I'm saying from a UX perspective it's restrictive to block the user and I'd rather let them enter an invalid value then tell them it's above their balance. I'm not suggesting letting them submit the tx; the button should be disabled.

However, the pattern you've got here is the same as the rest of the app so it's ok. It just felt counter-intuitive when I played with it on the test link